salt/modules/boto3_sns.py

```diff
@@ -301,7 +302,7 @@ def set_subscription_attributes(SubscriptionArn, AttributeName, AttributeValue,
         return False
 
 
-def subscribe(TopicArn, Protocol, Endpoint, region=None, key=None, keyid=None, profile=None):
+def subscribe(region=None, key=None, keyid=None, profile=None, **args):
```

Comment:
This is going to cause an API change and should be put on a deprecation path to be removed later.

Since these are positional args, we don't want to break people's setups.

Reply:
Ah, it didn't register that this would cause an API change...

I was only thinking in terms of keyword args (which are ALL I use, in both CLI and state defs) but for people passing them as ordered arguments I definitely see your concern.....

The alternative would be to add another explicit Attributes arg, as opposed to eating everything with **args.  This has the advantage of not breaking compat, but then it becomes a maintenance chore to add every new paramter as the underlying boto3 API expands...

This is exactly why I always use **args style in my own from-scratch boto modules :)

Anywho, I'll definitely follow up a.s.a.p. with your recommendation :)

Reply:
@rallytime 
What should I put in for the deprecation release version?

Reply:
@rallytime  grepping around in the code, it looked like Sodium was the likely choice here.  Let me know if something else is more appropriate :)